### ISSUE TITLE
Fix golangci-lint warnings

### ISSUE THREAD

**Issue by erikgrinaker** [tendermint/tendermint#4447]
`master` currently throws lots of linter warnings with the latest `golangci-lint`, we should fix these.

**Comment by erikgrinaker** [comment]
The remaining linter warnings are all caused by returning unexported types, e.g. this:

```
lite/multiprovider.go:19:56: exported func NewMultiProvider returns unexported type *github.com/tendermint/tendermint/lite.multiProvider, which can be annoying to use (golint)
func NewMultiProvider(providers ...PersistentProvider) *multiProvider {
```

I believe these are mostly used to force callers to use constructors instead of instantiating raw structs, but this prevents callers from e.g. placing the returned value in a struct or returning it themselves. We should either return interfaces, or export the structs.

**Comment by erikgrinaker** [comment]
Fixed by 8f48c49, b0683b7, and fe11219.
